dockerfile-image-update/src/main/java/com/salesforce/dockerfileimageupdate/utils/GithubAppCheck.java

```diff
@@ -120,40 +132,24 @@ protected boolean isGithubAppEnabledOnRepositoryWithGitApi(String fullRepoName)
      * Reference: https://github.com/mend/renovate-ce-ee/blob/main/docs/reporting-apis.md#repo-info
      */
     protected boolean isGithubAppEnabledOnRepositoryWithRenovateApi(String fullRepoName) {
-        try {
+        try (CloseableHttpClient httpClient = HttpClients.createDefault()) {
             String apiEndpoint = appServerApiEndpoint + "/api/repos/" + fullRepoName;
-            URL url = new URL(apiEndpoint);
-            HttpURLConnection conn = (HttpURLConnection) url.openConnection();
-            conn.setRequestMethod("GET");
-            conn.setRequestProperty("Accept", "application/json");
-            conn.setRequestProperty("authorization", "Bearer " + appServerApiToken);
-            conn.connect();
-            
-            Integer responseCode = conn.getResponseCode();
-            if (responseCode != 200) {
-                throw new RuntimeException("HttpResponseCode: " + responseCode);
+            HttpGet httpGet = new HttpGet(apiEndpoint);
+            httpGet.setHeader("Authorization", appServerApiToken);
+            httpGet.setHeader("Accept", "application/json");
+
+            HttpResponse response = httpClient.execute(httpGet);
+            int statusCode = response.getStatusLine().getStatusCode();
+            if (statusCode == 200) {
+                return true;
+            } else if (statusCode == 404) {
+                return false;
             } else {
-                String inline = "";
-                Scanner scanner = new Scanner(url.openStream());
-                while (scanner.hasNext()) {
-                    inline += scanner.nextLine();
-                }
-                scanner.close();
-
-                //Using the JSON simple library parse the string into a json object
-                JSONParser parse = new JSONParser();
-                JSONObject dataObject = (JSONObject) parse.parse(inline);
-                String appInstallationState = (String) dataObject.get("state");
-                String appActivationStatus = (String) dataObject.get("status");
-
-                // Return app installation and activation status
-                if (appInstallationState == "installed" && appActivationStatus == "activated") {
-                    return true;
-                }
+                log.warn("[isGithubAppEnabledOnRepositoryWithRenovateApi] -- Unexpected response code `{}` while trying to get app installation. Defaulting to False.", statusCode);
                 return false;
             }
-        } catch (Exception e) {
-            e.printStackTrace();
+        } catch (IOException exception) {
+            log.warn("[isGithubAppEnabledOnRepositoryWithRenovateApi] -- Caught a IOException {} while trying to get app installation. Defaulting to False", exception.getMessage());
```

Comment:
When logging exceptions we should include the full stack trace and the input parameters that could have caused the exception.

Reply:
@cjneasbi Done